server/src/compat/message_conversion/snapshots/retained_batch_snapshot.rs

```diff
@@ -77,56 +77,56 @@ impl Extendable for RetainedMessageBatchSnapshot {
 }
 
 impl TryFrom<Bytes> for RetainedMessageBatchSnapshot {
-    type Error = ServerError;
+    type Error = ServerCompatError;
     fn try_from(value: Bytes) -> Result<Self, Self::Error> {
         let base_offset = u64::from_le_bytes(
             value
                 .get(0..8)
-                .ok_or_else(|| {
-                    ServerError::CannotReadMessageBatchFormatConversion(
-                        "Failed to read batch base offset".to_owned(),
-                    )
-                })?
+                .ok_or_else(
+                    || ServerCompatError::CannotReadMessageBatchFormatConversion {
```

Comment:
I think we can also dip those `reason` fields, those were used as the `context` for our previous error type, now when we will add traces, we will put those in there instead.

Reply:
yep, I thought about that, but initially decided to leave it as is. However, I like the idea of removing the reason field and adding the trace instead

Reply:
done